geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/stats/MetadataBackedStats.scala

```diff
@@ -217,9 +215,12 @@ abstract class MetadataBackedStats(ds: DataStore, metadata: GeoMesaMetadata[Stat
     val stAttributesBuilder = Seq.newBuilder[String]
     val indexedAttributesBuilder = Seq.newBuilder[String]
 
-    sft.getIndices.foreach {
-      case i if Seq(Z3Index, XZ3Index, Z2Index, XZ2Index).exists(_.name == i.name) => stAttributesBuilder ++= i.attributes
-      case i => i.attributes.headOption.foreach(indexedAttributesBuilder += _)
+    sft.getIndices.foreach { i =>
+      if (i.attributes.headOption.contains(sft.getGeomField)) {
+        stAttributesBuilder ++= i.attributes
```

Comment:
What happens if someone tries to specify an attribute index with a geometry first and then some other index second?

Reply:
i'm pretty sure attribute indices don't support geometry types, so you'd get an error on createSchema if you tried that